build-plugins/kmp-conventions/src/main/kotlin/aws/sdk/kotlin/gradle/kmp/ConfigureTargets.kt

```diff
@@ -95,24 +97,28 @@ fun Project.configureKmpTargets() {
             }
         }
 
-        // enable targets
+        // enable the targets
         configureCommon()
 
         if (hasJvm && JVM_ENABLED) {
             configureJvm()
         }
 
         // FIXME Configure JS
-        // FIXME Configure Apple
-        // FIXME Configure Windows
 
-        withIf(hasLinux && NATIVE_ENABLED, kmpExt) {
-            configureLinux()
-        }
-
-        withIf(hasDesktop && NATIVE_ENABLED, kmpExt) {
-            configureLinux()
-            // FIXME Configure desktop
+        if (NATIVE_ENABLED) {
+            if (hasApple) {
+                kmpExt.apply { configureApple() }
+            }
+            if (hasWindows) {
+                kmpExt.apply { configureWindows() }
+            }
+            if (hasLinux) {
+                kmpExt.apply { configureLinux() }
+            }
+            if (hasDesktop) {
+                kmpExt.apply { configureWindows() }
```

Comment:
Question: `hasDesktop && NATIVE_ENABLED` resulting in `configureLinux()` before was wrong/temporary?

Reply:
Yeah good catch. It should be configuring all the desktop platforms (linux, macos, windows)